Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DefaultDartFeedbackBuilder.java

```diff
@@ -29,21 +29,19 @@ public void sendFeedback(@Nullable Project project, @Nullable String errorMessag
     String template = DartBundle.message("dart.feedback.url.template", ijBuild, sdkVsn, platDescr);
     if (errorMessage != null) {
       errorMessage = "```\n" + errorMessage + "```";
-      String potentialTemplate = template + "\n\n" + errorMessage;
-      if (potentialTemplate.length() <= MAX_URL_LENGTH) {
-        template = potentialTemplate;
-      }
-      else {
-        try {
-          File file = FileUtil.createTempFile("report", ".txt");
-          FileUtil.writeToFile(file, errorMessage);
-          potentialTemplate =
-            template + "\n\n" + DartBundle.message("dart.error.file.instructions", file.getAbsolutePath()) + "\n\n" + errorMessage;
-          template = potentialTemplate.substring(0, MAX_URL_LENGTH);
-        }
-        catch (IOException e) {
-          // ignore it
+      try {
+        File file = FileUtil.createTempFile("report", ".txt");
+        FileUtil.writeToFile(file, errorMessage);
+        if (serverLog != null) {
+          // Assume serverLog is never long enough that opening and closing the file is cheaper than copying it.
+          FileUtil.writeToFile(file, "\n\n" + serverLog, true);
```

Comment:
I think we can calculate contents first and then write it to a file in one shot.